lib/jsonapi-serializers/serializer.rb

```diff
@@ -252,7 +252,17 @@ def self.find_serializer(object, options)
       find_serializer_class(object, options).new(object, options)
     end
 
+    def self.proxy_objects(objects)
+      return nil unless objects
+      if objects.respond_to?(:map)
+        objects.map { |obj| DynamicProxyObject.new(obj) }
+      else
```

Comment:
IMO `respond_to?(:map)` is too weak of a trait check. Any object could have the method `map` and not be a collection.